package.json

```diff
@@ -15,10 +15,10 @@
     "semver"
   ],
   "scripts": {
-    "test": "npm run build && npm run lint && mocha -r ts-node/register test/index.spec.ts",
+    "test": "npm run build && npm run lint && ./automation/test.js",
```

Comment:
What does having the mocha script in a separate file add?

Unless there's a nice benefit, I'd rather keep it simple and do it inline here (might be nice to change the path to `test/*.spec.ts` though)

Reply:
The doc generation relies on knowing some information about how the mocha test runs, rather than parsing the test script with a regex (like we did before) I've made mocha run programmatically and use a shareable configuration file. It might seem silly to do this since it's only sharing a test target at the moment, however, I anticipate there being a point in the future where we want to add more complex options to mocha and it breaks the docs generation.